src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Filters/AddDependencyContext.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.Snapshot.Filters
+{
+    internal sealed class AddDependencyContext : IAddDependencyContext
+    {
+        private readonly ImmutableDictionary<string, IDependency>.Builder _worldBuilder;
+
+        private bool _acceptedOrRejected;
+        private IDependency _acceptedDependency;
+
+        public bool Changed { get; private set; }
+
+        public AddDependencyContext(ImmutableDictionary<string, IDependency>.Builder worldBuilder)
+        {
+            _worldBuilder = worldBuilder;
+        }
+
+        public void Initialize()
+        {
+            Changed = false;
+            _acceptedOrRejected = false;
+            _acceptedDependency = null;
+        }
+
+        public IDependency GetResult(IDependenciesSnapshotFilter filter)
+        {
+            if (!_acceptedOrRejected)
+            {
+                throw new InvalidOperationException(
+                    $"Filter type {filter.GetType()} must either accept or reject the dependency.");
+            }
+
+            _acceptedOrRejected = false;
+            return _acceptedDependency;
+        }
+
+        #region IAddDependencyContext
```

Comment:
We're a region free zone.

https://github.com/dotnet/project-system/blob/master/docs/repo/coding-conventions.md#code